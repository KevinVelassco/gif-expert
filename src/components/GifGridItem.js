import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card  animate__animated animate__flip">
            <div className="card__imgContainer">
                <img src={url} alt={title} className="card__img" />
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired

}
