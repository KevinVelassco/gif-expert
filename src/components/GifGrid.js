import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

    useEffect(() => {
        getGifs(category)
            .then(setimages)
    }, [category]);

    return (
        <>
            <h2>{category}</h2>
            <div className="panel">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img} />
                    ))
                }
            </div>
        </>
    );
}