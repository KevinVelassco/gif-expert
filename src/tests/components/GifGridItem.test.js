import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Test component <GifGridItem />', () => {

    const title = 'title imagen';
    const url = 'https://imagen.png';

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('El atributo src del <img/> debe tener como valor el prop url', () => {
        const img = wrapper.find('img');
        expect(img.prop("src")).toBe(url);
    });

    test('El atributo alt del <img/> debe tener como valor el prop title', () => {
        const img = wrapper.find('img');
        expect(img.prop("alt")).toBe(title);
    });

    test('El contenedor de la etiqueta <img/> debe tener className card__imgContainer', () => {
        const container = wrapper.find('img').parent();

        expect(container.hasClass("card__imgContainer")).toBe(true);

    });
});