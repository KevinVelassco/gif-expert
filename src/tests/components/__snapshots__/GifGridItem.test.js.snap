// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test componet <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card  animate__animated animate__flip"
>
  <div
    className="card__imgContainer"
  >
    <img
      alt="title imagen"
      className="card__img"
      src="https://imagen.png"
    />
  </div>
</div>
`;
